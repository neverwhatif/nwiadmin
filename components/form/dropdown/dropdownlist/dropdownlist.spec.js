import React from 'react';
import { shallow } from 'enzyme';

import DropdownList from './index';

jest.mock('nwiadmin/utility');

const data = [{
    id: 1,
    name: 'Lorem',
}, {
    id: 2,
    name: 'Ipsum',
}];

describe('DropdownList', () => {
    const component = shallow(<DropdownList data={data} />);
    const componentIsOpen = shallow(<DropdownList data={data} isOpen />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
        expect(componentIsOpen).toMatchSnapshot();
    });
    it('should be closed', () => {
        expect(component.type()).toBeNull();
    });
    it('should be open', () => {
        expect(componentIsOpen.type()).toBe('ul');
    });
    it('should handle onItemClick', () => {
        const mockOnClick = jest.fn();
        const componentWithClick = shallow(<DropdownList data={data} onItemClick={() => mockOnClick()} isOpen />);

        componentWithClick.find('li').first().simulate('click');
        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
    it('should handle onKeyDown', () => {
        const mockOnClick = jest.fn();
        const componentWithClick = shallow(<DropdownList data={data} onItemClick={() => mockOnClick()} isOpen />);

        componentWithClick.find('li').first().simulate('keyDown');
        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
});
