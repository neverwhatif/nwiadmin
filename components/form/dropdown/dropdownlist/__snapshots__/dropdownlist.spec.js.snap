// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownList should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropdownList
    data={
      Array [
        Object {
          "id": 1,
          "name": "Lorem",
        },
        Object {
          "id": 2,
          "name": "Ipsum",
        },
      ]
    }
    isOpen={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="item"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Lorem
        </li>,
        <li
          className="item"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Ipsum
        </li>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
          "className": "item",
          "onClick": [Function],
          "onKeyDown": [Function],
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "Ipsum",
          "className": "item",
          "onClick": [Function],
          "onKeyDown": [Function],
        },
        "ref": null,
        "rendered": "Ipsum",
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="item"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Lorem
          </li>,
          <li
            className="item"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Ipsum
          </li>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
            "className": "item",
            "onClick": [Function],
            "onKeyDown": [Function],
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "Ipsum",
            "className": "item",
            "onClick": [Function],
            "onKeyDown": [Function],
          },
          "ref": null,
          "rendered": "Ipsum",
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`DropdownList should render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropdownList
    data={
      Array [
        Object {
          "id": 1,
          "name": "Lorem",
        },
        Object {
          "id": 2,
          "name": "Ipsum",
        },
      ]
    }
    isOpen={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="item"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Lorem
        </li>,
        <li
          className="item"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Ipsum
        </li>,
      ],
      "className": "root rootIsOpen",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
          "className": "item",
          "onClick": [Function],
          "onKeyDown": [Function],
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "Ipsum",
          "className": "item",
          "onClick": [Function],
          "onKeyDown": [Function],
        },
        "ref": null,
        "rendered": "Ipsum",
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="item"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Lorem
          </li>,
          <li
            className="item"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Ipsum
          </li>,
        ],
        "className": "root rootIsOpen",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
            "className": "item",
            "onClick": [Function],
            "onKeyDown": [Function],
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "Ipsum",
            "className": "item",
            "onClick": [Function],
            "onKeyDown": [Function],
          },
          "ref": null,
          "rendered": "Ipsum",
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
